### PR TITLE
Refactor FXIOS-5286 [v109] Remove BVC dependence in TabScrollController & ReaderModeBarView

### PR DESCRIPTION
**Pull request by nishant2718**
# [FXIOS-5268](https://mozilla-hub.atlassian.net/browse/FXIOS-5286)

TLDR: Removes BVC dependence from TabScrollController & ReaderModeBarView.

## Details, details & details
It's not DRY. A couple options I thought of were:
- Move FeatureFlagsManager to AppContainer, then add that as a dependency to these places & do the same thing. Yes, maybe it would be ideal to do that and worth the effort eventually. But we can still achieve the same result with the way things are.
- Have `isBottomSearchBar` be a property in AppContainer. But then, we'd have a dependency w/i a dependency, and this one being a singleton. 

If it's Nimbus related, in general I'm a little apprehensive (and also open to ideas). If you guys have suggestions, hit me up!

### CODE REVIEW

**Review by adudenamedruby — APPROVED** [review]
I think this is ok for now. Like we talked about, I'm happy to move the `FeatureFlagManager` in the AppContainer, but let's look into that first and make sure nothing gets affected in ways we don't expect.

File: Client/Frontend/Browser/TabScrollController.swift [hunk 1]
```diff
@@ -103,8 +104,24 @@ class TabScrollingController: NSObject, FeatureFlaggable {
     private var lastContentOffset: CGFloat = 0
     private var scrollDirection: ScrollDirection = .down
     private var toolbarState: ToolbarState = .visible
+
+    private var isSearchBarLocationFeatureEnabled: Bool {
```
[the remark this answers is not in the document]
Comment (nishant2718) [reply Client/Frontend/Browser/TabScrollController.swift:108]: That seems reasonable. I can look into that!

**Review by adudenamedruby — APPROVED** [review]
Very nice and much cleaner! Tyvm for the updates. :)

**Review by OrlaM — CHANGES_REQUESTED** [review]
Small change request, otherwise looks good.

**Review by OrlaM — APPROVED** [review]

**Comment by nishant2718** [comment]
I guess another flaky test. I ran all `ReaderViewTest` locally and it seems fine.
